lib/control/v1/kms.js

```diff
@@ -2,40 +2,45 @@
 
 const KMSProvider = require('../../providers/kms');
 
+/**
+ * AWS-SDK gives out weirdly formatted keys, so make them
+ * consistent with formatting across the rest of Tokend
+ *
+ * @param {Object} obj
+ * @returns {Object}
+ */
+const normalizeKeys = (obj) => Object.keys(obj).reduce((acc, key) => {
+  acc[key.toLowerCase()] = obj[key];
+
+  return acc;
+}, {});
+
 /**
  * Route handler for KMS secrets
- * @param {StorageService} storage
  * @returns {function}
  */
-function KMS(storage) {
+function KMS() {
   return (req, res, next) => {
-    // KMS doesn't require a token so we can just pass an empty string
-    storage.lookup('', req.body, KMSProvider)
-      .then((result) => {
-        if (result) {
-          // AWS-SDK gives out weirdly formatted keys, so make them
-          // consistent with formatting across the rest of Tokend
-          Object.keys(result).forEach((key) => {
-            result[key.toLowerCase()] = result[key];
-            if (key.toLowerCase() !== key) {
-              delete result[key];
-            }
-          });
-        }
-
-        // Check for both undefined and null. `!= null` handles both cases.
-        if (result != null && result.hasOwnProperty('plaintext')) { // eslint-disable-line eqeqeq
-          result.plaintext = Buffer.from(result.plaintext, 'base64').toString();
-        }
-
-        if (result != null && result.hasOwnProperty('correlation_id')) { // eslint-disable-line eqeqeq
```

Comment:
Previously, the `LeaseManager` was responsible for creating a correlation ID with each request. Do we want to do something like this to keep that behavior?
```javascript
const Correlation = require('../../utils/correlation');
res.correlationId = Correlation.create();
```

Reply:
Do we need it? Previously the `correlation_id` was used to log data moving through various parts of the system. We're now bypassing all of it And calling the provider's decrypt method directly.

Reply:
Good point. I was mostly thinking about it from an API standpoint of maintaining the output we had previously.

Reply:
Yeah that was why I didn't bother. We intentionally removed the correlation id from the response object previously. The goal was consistency.